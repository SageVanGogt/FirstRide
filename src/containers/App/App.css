.App {
  top: 0;
  margin: 0;
  padding: 0;
  height: 100vh;
  background-color: #585858;
}

.nav-container {
  position: fixed;
  height: 90px;
  top: 0;
  z-index: 3;
  background-color: #404040;  
}

.error-container {
  position: fixed;
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 3;
  top: calc(50% - 100px);
  bottom: calc(50% - 100px);
  right: calc(50% - 200px);
  left: calc(50% - 200px);
  height: 200px;
  width: 400px;
  border-radius: 8px;
  background-color: white;
  font-size: 2rem;
}

.error-close-btn {
  background-color: #DC0CE8;
  margin-top: 10px;
  border-radius: 3px;
}

.error-container h1 {
  position: absolute;
  width: 100%;
  margin: 0;
  padding: 5px;
  font-family: 'Work Sans', sans-serif;
  border-radius: 8px 8px 0 0;
  text-align: center;
  top: 0;
  background-color: grey;
}


