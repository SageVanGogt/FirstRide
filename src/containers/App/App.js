import React, { Component } from 'react';
import './App.css';
import { Route, Redirect, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import DestinationsContainer from './../DestinationsContainer/DestinationsContainer';
import RidesContainer from './../RidesContainer/RidesContainer';
import SigninContainer from './../SigninContainer/SigninContainer';
import ProfileContainer from './../ProfileContainer/ProfileContainer';
import NavContainer from './../NavContainer/NavContainer';
import LoginComponent from './../../components/LoginComponent/LoginComponent';
import * as actions from './../../actions/error';
import UserRidesContainer from './../../containers/UserRidesContainer/UserRidesContainer';
export class App extends Component {
  constructor() {
    super(),

    this.state = {
      showLogin: false,
      showUserRides: false
    }
  }

  toggleLogin = () => {
    this.setState({
      showLogin: !this.state.showLogin
    });
  };

  toggleShowUserRides = () => {
    if (!this.props.user.id) {
      this.props.setError('You must ')
      return;
    };
    this.setState({
      showUserRides: !this.state.showUserRides
    });
  };

  errorElement = () => {
    return (
      <div className="error-container">
        <h1>Whoops!</h1>
        <strong>{this.props.error}</strong>
        <button 
          onClick={this.props.removeError}
          className="error-close-btn">
          ok!
        </button>
      </div>
    );
  }

  render() {
    return (
      <div className="App">
        <div className="nav-container">
          <NavContainer 
            toggleLogin={this.toggleLogin}
            toggleShowUserRides={this.toggleShowUserRides}/>
        </div>
        {
          this.props.error.length &&
          this.errorElement()
        }
        {
          this.state.showUserRides &&
          <UserRidesContainer
          toggleShowUserRides={this.toggleShowUserRides} />
        }
        {
          this.state.showLogin & !this.props.user.id && 
          <LoginComponent
            toggleLogin={this.toggleLogin}/>
        }
        <Switch>
          <Route 
            exact path="/" 
            component={DestinationsContainer}
          />
          <Route 
            exact path="/rides" 
            component={RidesContainer}
          />
          <Route 
            exact path="/profile" 
            component={ProfileContainer}
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  destination: state.destination,
  error: state.error
});

export const mapDispatchToProps = (dispatch) => ({
  removeError: () => dispatch(actions.removeError()),
  setError: (error) => dispatch(actions.addError(error))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))
