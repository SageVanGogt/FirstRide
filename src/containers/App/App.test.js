import React from 'react';
import { App, mapStateToProps } from './App';
import { shallow } from 'enzyme';
 
describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should pull user from the state', () => {
      let mockState = {
        user: {
          email: "thurmanvogt@gmail.com"
        },
        destination: {
          location_name: 'Red Rocks'
        }
      };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.user;
      let expected = mockState.user;

      expect(actual).toEqual(expected);
    })

    it('should pull destination from the state', () => {
      let mockState = {
        user: {
          email: "thurmanvogt@gmail.com"
        },
        destination: {
          location_name: 'Red Rocks',
          id: 1
        },
        rides: []
      };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.destination;
      let expected = mockState.destination;

      expect(actual).toEqual(expected);
    })
  })
})
