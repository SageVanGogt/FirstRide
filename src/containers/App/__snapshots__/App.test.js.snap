// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Connect(NavContainer) />
</div>,
        <Switch>
          <Route
                    exact={true}
                    path="/signin"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    exact={true}
                    path="/rides"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/profile"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(NavContainer) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/signin"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              exact={true}
              path="/rides"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/profile"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signin",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/rides",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/profile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Connect(NavContainer) />
</div>,
          <Switch>
            <Route
                        exact={true}
                        path="/signin"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        exact={true}
                        path="/rides"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/profile"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(NavContainer) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/signin"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                exact={true}
                path="/rides"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/profile"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/signin",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/rides",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/profile",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
